plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -427,7 +442,7 @@ public static ImmutableList<DeltaLakeColumnMetadata> columnsWithStats(List<Delta
                 .collect(toImmutableList());
     }
 
-    private Stream<AddFileEntry> activeAddEntries(Stream<DeltaLakeTransactionLogEntry> checkpointEntries, List<Transaction> transactions)
+    private Stream<AddFileEntry> activeAddEntries(Stream<DeltaLakeTransactionLogEntry> checkpointEntries, List<Transaction> transactions, TrinoFileSystem fileSystem)
```

Comment:
To investigate whether we can do without passing `fileSystem` given that `TransactionLogEntries` holds already a pointer to `TrinoInputFile`

Reply:
See https://github.com/trinodb/trino/pull/24491#discussion_r1895342752